### PR TITLE
modules: thrift: update doc and makefiles to build with macos

### PR DESCRIPTION
**Pull request by cfriedt**
Added tabs to the thrift doc so that building works for macos and is consistent with the rest of the documentation in terms of one tab per os (I unfortunately do not have a windows dev environment at my disposal, but would be happy to add those package installation instructions if someone knows what they might be).

Updated client and server makefiles to discover include paths for boost and openssl.

[Doc Preview](https://builds.zephyrproject.io/zephyr/pr/74557/docs/samples/modules/thrift/hello/README.html)

### CODE REVIEW

File: samples/modules/thrift/hello/client/Makefile [hunk 1]
```diff
@@ -14,6 +16,14 @@ GENOBJ = $(filter-out %.h, $(GENSRC:.cpp=.o))
 THRIFT_FLAGS :=
 THRIFT_FLAGS += $(shell pkg-config --cflags thrift)
 THRIFT_FLAGS += -I$(GEN_DIR)
+ifeq ($(OS),Darwin)
+# get Homebrew prefix
+HOMEBREW_PREFIX := $(shell brew --prefix)
```
[the remark this answers is not in the document]
Comment (cfriedt) [reply samples/modules/thrift/hello/client/Makefile:21]: 😂 Well, that's one solution, I guess

**Review by kartben — APPROVED** [review]

File: samples/modules/thrift/hello/client/Makefile [hunk 2]
```diff
@@ -3,6 +3,8 @@
 
 .PHONY: all clean
 
+OS = $(shell uname -s)
```
Comment (tomi-font) [inline samples/modules/thrift/hello/client/Makefile:6]:
```suggestion
OS := $(shell uname -s)
```

File: samples/modules/thrift/hello/server/Makefile [hunk 3]
```diff
@@ -3,6 +3,8 @@
 
 .PHONY: all clean
 
+OS = $(shell uname -s)
```
Comment (tomi-font) [inline samples/modules/thrift/hello/server/Makefile:6]:
```suggestion
OS := $(shell uname -s)
```

File: samples/modules/thrift/hello/README.rst [hunk 4]
```diff
@@ -57,12 +57,23 @@ Requirements
...
+- Thrift dependencies installed for your host OS
 
-.. code-block:: console
-   :caption: Install additional dependencies in Ubuntu
+.. tabs::
+
+   .. group-tab:: Ubuntu
+
+      .. code-block:: bash
+        :caption: Install thrift dependencies in Ubuntu
```
Comment (tomi-font) [inline samples/modules/thrift/hello/README.rst:67]:
```suggestion
        :caption: Install thrift dependencies on Ubuntu
```

File: samples/modules/thrift/hello/README.rst [hunk 5]
```diff
@@ -57,12 +57,23 @@ Requirements
...
+
+         sudo apt install -y libboost-all-dev thrift-compiler libthrift-dev
+
+   .. group-tab:: macOS
+
+      .. code-block:: bash
+        :caption: Install thrift dependencies in macOS
```
Comment (tomi-font) [inline samples/modules/thrift/hello/README.rst:74]:
```suggestion
        :caption: Install thrift dependencies on macOS
```

**Review by tomi-font — APPROVED** [review]

File: samples/modules/thrift/hello/README.rst (hunk 5 above)
Comment (cfriedt) [reply samples/modules/thrift/hello/README.rst:74]: Hmm.. do people normally say a package is installed "on" an OS? I usually say "in".

It's possibly "on" in German or French, but I've been saying "in" for the last 20 years or so and I'm a native English speaker.

Comment (tomi-font) [reply samples/modules/thrift/hello/README.rst:74]: Oh. Could well be that you're right then. I say "on" but I'm not a native English speaker.

**Pull request merged**